src/tito/common.py

```diff
@@ -852,25 +852,35 @@ def get_commit_timestamp(sha1_or_tag):
 
 
 def create_tgz(git_root, prefix, commit, relative_dir,
-    dest_tgz):
+    dest_tgz, method='git', add_git_folder=False):
     """
     Create a .tar.gz from a projects source in git.
     """
     os.chdir(os.path.abspath(git_root))
     timestamp = get_commit_timestamp(commit)
 
-    # Accomodate standalone projects with specfile i root of git repo:
+    # Accommodate standalone projects with specfile in root of git repo:
     relative_git_dir = "%s" % relative_dir
-    if relative_git_dir in ['/', './']:
-        relative_git_dir = ""
+    if relative_git_dir in ['/', '']:
+        relative_git_dir = "./"
 
     basename = os.path.splitext(dest_tgz)[0]
     initial_tar = "%s.initial" % basename
 
     # command to generate a git-archive
-    git_archive_cmd = 'git archive --format=tar --prefix=%s/ %s:%s --output=%s' % (
-        prefix, commit, relative_git_dir, initial_tar)
-    run_command(git_archive_cmd)
+    if method == 'git':
+        git_archive_cmd = 'git archive --format=tar --prefix=%s/ %s:%s --output=%s' % (
+            prefix, commit, relative_git_dir, initial_tar)
+        run_command(git_archive_cmd)
+        if add_git_folder:
+            tar_append_cmd = f"tar --append -f {initial_tar}  --transform 's|^{relative_git_dir}|{prefix}/|' {relative_git_dir}/.git"
+            run_command(tar_append_cmd)
+    elif method == 'tar':
```

Comment:
I have some issues when using

```
[archiver]
method = tar
```

Traceback:

```
Traceback (most recent call last):
  File "/home/jkadlcik/git/tito/src/tito/cli.py", line 919, in <module>
    main()
  File "/home/jkadlcik/git/tito/src/tito/cli.py", line 910, in main
    CLI().main(sys.argv[1:])
  File "/home/jkadlcik/git/tito/src/tito/cli.py", line 209, in main
    return module.main(argv)
           ^^^^^^^^^^^^^^^^^
  File "/home/jkadlcik/git/tito/src/tito/cli.py", line 392, in main
    return builder.run(self.options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jkadlcik/git/tito/src/tito/builder/main.py", line 160, in run
    self.tgz()
  File "/home/jkadlcik/git/tito/src/tito/builder/main.py", line 577, in tgz
    self._setup_sources()
  File "/home/jkadlcik/git/tito/src/tito/builder/main.py", line 610, in _setup_sources
    create_tgz(self.git_root, self.tgz_dir, self.git_commit_id,
  File "/home/jkadlcik/git/tito/src/tito/common.py", line 895, in create_tgz
    tarfixer.fix()
  File "/home/jkadlcik/git/tito/src/tito/tar.py", line 325, in fix
    self.process_chunk(chunk)
  File "/home/jkadlcik/git/tito/src/tito/tar.py", line 292, in process_chunk
    chunk_props[member] = int(chunk_props[member], 8)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 8: ''
```

Reproducible when you run `tito build --tgz` within this project.